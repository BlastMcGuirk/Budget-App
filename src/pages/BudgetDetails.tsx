import { NativeStackScreenProps } from '@react-navigation/native-stack';
import React, { useMemo } from 'react';
import { View, ScrollView, Text, StyleSheet } from 'react-native';
import { RootStackParamList } from '../App';
import BudgetSummary from '../components/BudgetSummary';
import ListItem from '../components/ListItem';
import { Budget } from '../interfaces/Budget';
import { Item } from '../interfaces/Item';
import { FontSizes } from '../styles/global';

export interface BudgetDetailsProps {
    budget: Budget;
    items: Item[];
}

type Props = NativeStackScreenProps<RootStackParamList, 'BudgetDetails'>;

export default function BudgetDetails(props: Props) {
    const { budget, items } = props.route.params;
    const sum = useMemo(() => {
        return items.reduce((acc, cur) => acc + cur.amount, 0);
    }, items);
    return (
        <ScrollView key={budget.id} contentContainerStyle={styles.container}>
            {/*<Text style={[FontSizes.XL, styles.title]}>{budget}</Text>*/}
            <View style={styles.header}>
                <BudgetSummary budget={budget.name} remaining={sum} total={budget.budgetValue} />
                <Text
                    style={[styles.new, FontSizes.S]}
                    onPress={() => console.log("Hello")}
                >
                    New Entry...
                </Text>
            </View>
            {items.map(item => {
                return <ListItem key={item.id} item={item}/>
            })}
        </ScrollView>
    )
}

const styles = StyleSheet.create({
    container: {
        flexGrow: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
    },
    header: {
        borderBottomWidth: 2,
        width: '100%',
    },
    new: {
        borderTopWidth: 1,
        paddingHorizontal: 16,
        paddingVertical: 8
    }
});