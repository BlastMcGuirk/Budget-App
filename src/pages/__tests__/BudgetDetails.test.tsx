import renderer from 'react-test-renderer';
import BudgetDetails, { BudgetDetailsProps } from '../BudgetDetails';

describe("Budget Details Page Tests", () => {

    it("Renders correctly", () => {
        const props: any = {
            route: {
                params: {
                    budget: 'needs',
                    items: [
                        {
                            name: 'Rent',
                            amount: 1757.25,
                            date: '8/4/2023',
                            category: 'Bills'
                        }
                    ]
                }
            },
        };
        const tree = renderer.
            create(<BudgetDetails {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});