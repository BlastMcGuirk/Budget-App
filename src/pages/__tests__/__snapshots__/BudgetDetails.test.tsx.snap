// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Budget Details Page Tests Renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flexGrow": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderBottomWidth": 2,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          NEEDS
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 36,
              },
              Object {
                "color": "green",
              },
            ]
          }
        >
          $
          1234.56
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          / $
          1500
        </Text>
      </View>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "borderTopWidth": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            },
            Object {
              "fontSize": 16,
            },
          ]
        }
      >
        New Entry...
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "gray",
          "borderBottomWidth": 1,
          "padding": 16,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "baseline",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Rent
        </Text>
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          $
          1757.25
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "baseline",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          Bills
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          8/4/2023
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
